src/server/internalServer.cpp

```diff
@@ -1004,11 +1004,11 @@ std::unique_ptr<Response> InternalServer::handle_search_request(const RequestCon
     return response;
   }
 
-  const auto start = max(1u, request.get_optional_param("start", 1u));
+  const auto start = max(0u, request.get_optional_param("start", 0u));
```

Comment:
This changes the semantics of the `start` parameter of the pagination control of the [`/search` API endpoint of `kiwix-serve`](https://kiwix-tools.readthedocs.io/en/latest/kiwix-serve.html#search) from 1-based to 0-based. Either the bug should be fixed in a different way or the documentation in `kiwix/kiwix-tools` has to be updated, too.

Reply:
Yes, this is on purpose and normal. The query is a technical thing and the labelling is for users. It should always have been like this.

Reply:
OK, then go ahead with fixing the tests.